simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -62,7 +63,18 @@
    * @param address  host:port or ip:port of the Pushgateway.
    */
   public PushGateway(String address) {
-    this.address = address;
+    this(URI.create("http://" + address));
+  }
+
+  /**
+   * Construct a Pushgateway, with the given URI.
+   * <p>
+   * @param serverBaseURI the base URI and optional context path of the Pushgateway server.
+   */
+  public PushGateway(URI serverBaseURI) {
+    this.address = URI.create(serverBaseURI.toString() + "/metrics/job/")
```

Comment:
Thanks! I've bypassed this problem by constructing a `URI` instance based on the `URL` that's passed to this constructor.